### PR TITLE
Add optional sensitivity computation in ExcessMapEstimator

### PR DESCRIPTION
**Pull request by QRemy**
Add optional sensitivity computation in ExcessMapEstimator

### CODE REVIEW

File: gammapy/estimators/map/excess.py [hunk 1]
```diff
@@ -127,13 +130,14 @@ class ExcessMapEstimator(Estimator):
...
     tag = "ExcessMapEstimator"
-    _available_selection_optional = ["errn-errp", "ul"]
+    _available_selection_optional = ["errn-errp", "ul", "sensitivity"]
 
     def __init__(
         self,
         correlation_radius="0.1 deg",
         n_sigma=1,
         n_sigma_ul=2,
+        n_sigma_sensitivity=4,
```
Comment (bkhelifi) [inline gammapy/estimators/map/excess.py:140]: It is 5 normally. Why 4?

File: gammapy/estimators/map/excess.py [hunk 2]
```diff
@@ -86,13 +86,16 @@ class ExcessMapEstimator(Estimator):
         Confidence level for the asymmetric errors expressed in number of sigma.
     n_sigma_ul : float
         Confidence level for the upper limits expressed in number of sigma.
+    n_sigma_sensitivity : float
+        Confidence level for the sensitivity expressed in number of sigma.
```
Comment (bkhelifi) [inline gammapy/estimators/map/excess.py:90]: There are other options needed when computing the sensitivity, the minimal number of gamma-rays and the fraction of background counts above which the number of gamma-rays is (ie eta for which Ng>eta*Nkg).

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy ! This is a nice idea. This can be a useful feature.

Maybe one general question, could this make sense in the broad context of flux estimation.

File: gammapy/estimators/map/core.py [hunk 3]
```diff
@@ -58,7 +58,7 @@
 OPTIONAL_QUANTITIES = {
     "dnde": ["dnde_err", "dnde_errp", "dnde_errn", "dnde_ul"],
     "e2dnde": ["e2dnde_err", "e2dnde_errp", "e2dnde_errn", "e2dnde_ul"],
-    "flux": ["flux_err", "flux_errp", "flux_errn", "flux_ul"],
+    "flux": ["flux_err", "flux_errp", "flux_errn", "flux_ul", "sensitivity"],
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/estimators/map/core.py:61]: changed to `flux_sensitivity`

File: gammapy/estimators/map/excess.py (hunk 2 above)
Comment (QRemy) [reply gammapy/estimators/map/excess.py:90]: I added `gamma_min_sensitivity`, `bkg_syst_fraction_sensitivity and `apply_threshold_sensitivity` options.
``bkg_syst_fraction_sensitivity` and `gamma_min_sensitivity` are used in the sensitivity computation only if  `apply_threshold_sensitivity` is True. The default is set to False which match the behaviour used for the HGPS catalogue.

File: gammapy/estimators/map/excess.py [hunk 4]
```diff
@@ -86,13 +86,25 @@ class ExcessMapEstimator(Estimator):
...
+    gamma_min_sensitivity : float, optional
+        Minimum number of gamma-rays. Default is 10.
+        Usde only for for sensitivity computation if `apply_threshold_sensitivity`is True.
+    bkg_syst_fraction_sensitivity : float, optional
+        Fraction of background counts above which the number of gamma-rays is . Default is 0.05
+        Usde only for for sensitivity computation if `apply_threshold_sensitivity`is True.
+    apply_threshold_sensitivity : bool
+        If True use ``bkg_syst_fraction_sensitivity` and `gamma_min_sensitivity`
+        in sensitivity computation. Defalut is False which is setting used for the HGPS catalogue.
```
Comment (registerrier) [inline gammapy/estimators/map/excess.py:99]:
```suggestion
        in sensitivity computation. Default is False which is setting used for the HGPS catalogue.
```

File: gammapy/estimators/map/excess.py [hunk 5]
```diff
@@ -86,13 +86,25 @@ class ExcessMapEstimator(Estimator):
...
     n_sigma_ul : float
         Confidence level for the upper limits expressed in number of sigma.
+    n_sigma_sensitivity : float
+        Confidence level for the sensitivity expressed in number of sigma.
+    gamma_min_sensitivity : float, optional
+        Minimum number of gamma-rays. Default is 10.
+        Usde only for for sensitivity computation if `apply_threshold_sensitivity`is True.
+    bkg_syst_fraction_sensitivity : float, optional
+        Fraction of background counts above which the number of gamma-rays is . Default is 0.05
+        Usde only for for sensitivity computation if `apply_threshold_sensitivity`is True.
```
Comment (registerrier) [inline gammapy/estimators/map/excess.py:96]:
```suggestion
        Used only for for sensitivity computation if `apply_threshold_sensitivity`is True.
```

**Review by registerrier — APPROVED** [review]
thanks @QRemy ! Looks good!

**Pull request merged**